### PR TITLE
NYIF HTML certs customizations

### PR DESCRIPTION
**Pull request by bryanlandia**
These were originally made against nyif/feature/mergeDogwood.1, which is still shown as the active edx-platform version for NYIF production in server-vars.yml. Prod server is actually running nyif/release though.  :-{  

These changes are not yet approved for deployment to prod or merge to nyif/release, though.

### CODE REVIEW

**Comment by bryanlandia** [comment]
These changes interact with the custom certs code in `appsembleredx` package which at this point is only deployed to the merge-cca (staging).

**Comment by abeals** [comment]
[(2) Enabled certs for NYIF in production](https://trello.com/c/6Y5wLIgr/582-2-enabled-certs-for-nyif-in-production)

File: cms/djangoapps/models/settings/course_metadata.py [hunk 1]
```diff
@@ -121,7 +121,9 @@ def fetch_all(cls, descriptor):
                 'value': field.read_json(descriptor),
                 'display_name': _(field.display_name),    # pylint: disable=translation-of-non-string
                 'help': _(field.help),                    # pylint: disable=translation-of-non-string
-                'deprecated': field.runtime_options.get('deprecated', False)
+                'deprecated': field.runtime_options.get('deprecated', False),
+                'values': getattr(field, 'values', None) if type(field).__name__ not in ("Boolean") and type(getattr(field, 'values', None)).__name__ in ('list', 'tuple') else None,
```
Comment (johnbaldwin) [inline cms/djangoapps/models/settings/course_metadata.py]: Small nit on styling: line is too long, hard to read.

Also, Python is friendly about last item having a comma, this way we can add to a list w/out showing a code change on that line

File: cms/static/js/views/settings/advanced.js [hunk 2]
```diff
@@ -9,17 +9,21 @@ var AdvancedView = ValidatingView.extend({
...
-        'blur :input' : "blurInput"
+        'blur :input' : "blurInput",
+        "change select": "clearValidationErrors",
+        "change select": "selectSetField"
+
         // TODO enable/disable save based on validation (currently enabled whenever there are changes)
     },
     initialize : function() {
-        this.template = _.template($("#advanced_entry-tpl").text());
+        this.base_template = _.template($("#advanced_entry-tpl").text());
```
Comment (johnbaldwin) [inline cms/static/js/views/settings/advanced.js]: @bryanlandia Where is the template that contains these IDs?

File: lms/djangoapps/certificates/tests/test_webview_views.py [hunk 3]
```diff
@@ -277,7 +276,7 @@ def test_rendering_maximum_data(self):
         )
         # Test an item from user info
         self.assertIn(
-            "{fullname}, you've earned a certificate!".format(fullname=self.user.profile.name),
+            "{fullname}, you&#39;ve earned a certificate!".format(fullname=self.user.profile.name),
```
Comment (johnbaldwin) [inline lms/djangoapps/certificates/tests/test_webview_views.py]: Why not say "you have"?

File: cms/static/js/views/settings/advanced.js [hunk 4]
```diff
@@ -157,6 +163,19 @@ var AdvancedView = ValidatingView.extend({
...
+    },
+    selectSetField : function(event) {
+        var self = this;
+        var key = this.selectorToField[event.currentTarget.id];
+        var newVal = $(event.currentTarget).val();
+        var JSONValue = JSON.parse(newVal);
+        var modelVal = self.model.get(key);
+        modelVal.value = JSONValue;
+        self.model.set(key, modelVal);
+        var message = gettext("Your changes will not take effect until you save your progress. Take care with key and value formatting, as validation is not implemented.");
```
Comment (johnbaldwin) [inline cms/static/js/views/settings/advanced.js]: Long line

**Pull request merged**